mlflow/utils/oss_utils.py

```diff
@@ -0,0 +1,34 @@
+import urllib.parse
+
+from mlflow.exceptions import MlflowException
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.rest_utils import MlflowHostCreds
+from mlflow.utils.uri import (
+    _DATABRICKS_UNITY_CATALOG_SCHEME,
+)
+
+
+def get_oss_host_creds(server_uri=None):
```

Comment:
Could we make this use a short-circuit approach? Something along the lines of:

```python
def get_oss_host_creds(server_uri=None):
    """
    Retrieve the host credentials for the OSS server.
    
    Args:
        server_uri (str): The URI of the server.
    
    Returns:
        MlflowHostCreds: The host credentials for the OSS server.
    """

    parsed_uri = urllib.parse.urlparse(server_uri)
    
    # Short-circuit early
    if parsed_uri.scheme != "uc":
        raise MlflowException("The scheme of the server_uri should be 'uc'")
    
    if parsed_uri.path == _DATABRICKS_UNITY_CATALOG_SCHEME:
        return get_databricks_host_creds(parsed_uri.path)
    
    return MlflowHostCreds(
        host=f"{parsed_uri.path}"
    )
```

Reply:
Thanks for the suggestion. I've done this instead now.